### ISSUE TITLE
sentinels option must be symbolized

### ISSUE THREAD

**Issue by fjyaniez** [redis/redis-rb#570]
Hi,

Loading the configuration from a YML file I noticed the following inconsistent behaviour while initializing `Redis`, depending on passing the sentinels parameters as a key or as a symbol:

``` ruby
irb(main):004:0> redis = Redis.new("url"=>"redis://mymaster", "sentinels"=>[{"host"=>"slave1", "port"=>6381}, {"host"=>"slave2", "port"=>6381}])
=> #<Redis client v3.2.1 for redis://mymaster:6379/0>
irb(main):005:0> redis.set 'key', 'key'
SocketError: getaddrinfo: Name or service not known
```

``` ruby
irb(main):006:0> redis = Redis.new("url"=>"redis://mymaster", :sentinels=>[{"host"=>"slave1", "port"=>6381}, {"host"=>"slave2", "port"=>6381}])
=> #<Redis client v3.2.1 for redis://mymaster:6379/0>
irb(main):007:0> redis.set 'key', 'key'
TypeError: can't convert nil into Integer
```

``` ruby
irb(main):010:0> redis = Redis.new("url"=>"redis://mymaster", :sentinels=>[{"host"=>"slave1", :port=>6381}, {"host"=>"slave2", :port=>6381}])
=> #<Redis client v3.2.1 for redis://mymaster:6379/0>
irb(main):011:0> redis.set 'key', 'key'
SocketError: getaddrinfo: Name or service not known
```

``` ruby
irb(main):012:0> redis = Redis.new("url"=>"redis://mymaster", :sentinels=>[{:host=>"slave1", :port=>6381}, {:host=>"slave2", :port=>6381}])
=> #<Redis client v3.2.1 for redis://mymaster:6379/0>
irb(main):013:0> redis.set 'key', 'key'
=> "OK"
irb(main):014:0>
```

``` ruby
irb(main):012:0> redis = Redis.new("url"=>"redis://mymaster", "sentinels"=>[{:host=>"slave1", :port=>6381}, {:host=>"slave2", :port=>6381}])
=> #<Redis client v3.2.1 for redis://mymaster:6379/0>
irb(main):013:0> redis.set 'key', 'key'
SocketError: getaddrinfo: Name or service not known
```

Would be safe to use an equivalent of Rail's `deep_symblize_keys` method on `Redis::Client`'s `initialize` method to recursively symbolize the `options` parameter?

Notice for example that you can use `'url'` or `:url` as key in the parameters indifferently.

**Comment by dgtv** [comment]
+1

**Comment by drewish** [comment]
I also ran into this. Happy to do a PR if this is something the maintainers are into changing.

**Comment by fjyaniez** [comment]
@drewish I created the PR almost a year ago, this project seems to be abandoned :(

**Comment by drewish** [comment]
Ah I'd totally missed the link to the PR

**Comment by skippy** [comment]
the other challenge seems to be if you don't specify a port, you also get the `TypeError: can't convert nil into Integer` error.  Once port is specified, with symbolized keys of course, this works.

**Comment by byroot** [comment]
Fixed in #599
